Pr 9180

todb-r7 opened the pull request:
Updated after @bcoles's review over on rapid7/metasploit-framework#9180

One more test, @RootUp ! Mostly to check that the conditionals actually fall through correctly.

After that, I'll muddle through the markdown documentation, unless you want to give that a shot, as described here:

https://github.com/rapid7/metasploit-framework/wiki/Generating-Module-Documentation

Thanks for your patience on this.

Commit:
Nits picked by @bcoles: commas, quotes, and <head>

Commit:
Better conditionals from @bcoles

bcoles commented on modules/auxiliary/gather/samsung_browser_sop_bypass.rb:
```diff
@@ -118,11 +119,10 @@ def collect_data(request)
     p = cred['pass']
-    if not u.nil? and not u.empty? and not p.nil? and not p.empty?
-      print_good("#{cli.peerhost}: Collected credential for '#{datastore['TARGET_URL']}' #{u}:#{p}")
-      store_cred(u,p)
```
Was `store_cred(u,p)` supposed to have been removed ?

RootUp commented:
Hi @todb-r7 and @bcoles 
While reload_all it generates the below error.

`
msf > reload_all 
[*] Reloading modules from all module paths...
[-] WARNING! The following modules could not be loaded!
[-] 	/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb: SyntaxError /opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:52: syntax error, unexpected tIDENTIFIER, expecting ']'
...) function. Use the variable 'x' to refer to the new tab.',
...                               ^
/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:54: syntax error, unexpected ']', expecting keyword_end
    ])
     ^
/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:143: syntax error, unexpected end-of-input, expecting keyword_end
`

todb-r7 commented:
Oops, yeah I screwed it up. My fault for not testing before pushing!

Will fix in a bit.

Commit:
And this is why I hate using single quotes

Also, restored the store_cred call.

This will fix up RootUp/metasploit-framework#3 for PR #9180

todb-r7 commented:
Fixed, @RootUp !
